driver-core/src/main/com/mongodb/internal/operation/CreateCollectionOperation.java

```diff
@@ -341,6 +350,16 @@ public CreateCollectionOperation collation(final Collation collation) {
         return this;
     }
 
+    public CreateCollectionOperation expireAfter(@Nullable final Duration expireAfter) {
+        this.expireAfter = expireAfter;
+        return this;
+    }
+
+    public CreateCollectionOperation timeSeriesOptions(@Nullable final TimeSeriesOptions timeSeriesOptions) {
```

Comment:
This breaks the previous immutability rule.

Reply:
You mean about operations?  I don't think we really have immutability of operations.  Consider that we often do shallow copies of BSON documents, which applications can maintain references to.

Reply:
Agreed